lib/PipelineC/CMakeLists.txt

```diff
@@ -2,6 +2,34 @@
 # This file is distributed under the MIT License. See LICENSE.md for details.
 #
 
-revng_add_library_internal(revngPipelineC SHARED PipelineC.cpp)
+add_custom_command(
+  OUTPUT "${CMAKE_BINARY_DIR}/lib/PipelineC/types.inc"
+         "${CMAKE_BINARY_DIR}/lib/PipelineC/functions.inc"
+         "${CMAKE_BINARY_DIR}/lib/PipelineC/wrappers.inc"
+  COMMAND
+    # cmake-format: off
+    python "${CMAKE_SOURCE_DIR}/scripts/PipelineC_add_tracing.py"
+    -i "${CMAKE_SOURCE_DIR}/include/revng/PipelineC/ForwardDeclarationsC.h"
+    -i "${CMAKE_SOURCE_DIR}/include/revng/PipelineC/Prototypes.h"
+    "${CMAKE_BINARY_DIR}/lib/PipelineC"
+    # cmake-format: on
+  DEPENDS "${CMAKE_SOURCE_DIR}/scripts/PipelineC_add_tracing.py"
+          "${CMAKE_SOURCE_DIR}/include/revng/PipelineC/ForwardDeclarationsC.h"
+          "${CMAKE_SOURCE_DIR}/include/revng/PipelineC/Prototypes.h")
+
+revng_add_library_internal(
+  revngPipelineC SHARED "${CMAKE_BINARY_DIR}/lib/PipelineC/types.inc"
+  "${CMAKE_BINARY_DIR}/lib/PipelineC/wrappers.inc" Impl/PipelineC.cpp)
```

Comment:
Why?

Reply:
Without those CMake never invokes the command to generate the `.inc` files

Reply:
1. Dropped, it was needed for `clang-tidy`, I guess we can defer this once we're on LLVM 16 and can put `// NOLINTBEGIN` in `PipelineC.cpp` and drop the `.clang-tidy` file
2. Switched to adding a target, dunno why CMake wouldn't add the command to the dependency tree
3. Switched the file names accordingly